### PR TITLE
Update make_future and train_test_split regressors handling

### LINKED ISSUE #443: Improve regressors handling in TSDataset
### 🚀 Feature Request

Make `make_future` and `train_test_split` pass the list of regressors to the generated datasets.

### Motivation

In #327 we added parameter `known_future` in the constructor of TSDataset. However methods `make_future` and `train_test_split` don't use this parameter constructing the output.

### Proposal

1. Pass self.regressors to the constructor in `make_future` here
2. Pass self.regressors to the constructor in `train_test_split` here and here
3. Create test for this cases  

### Test cases

1. Check that future part inherit the list of regressor from dataset after applying `make_future` 
2. Check that train and test parts inherit the list of regressor from dataset after applying `train_test_split`

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [x] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by alex-hse-repository**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Closing issues

closes #443

### CODE REVIEW

File: etna/datasets/tsdataset.py [hunk 1]
```diff
@@ -278,7 +278,9 @@ def make_future(self, future_steps: int) -> "TSDataset":
 
         future_dataset = df.tail(future_steps).copy(deep=True)
         future_dataset = future_dataset.sort_index(axis=1, level=(0, 1))
-        future_ts = TSDataset(future_dataset, freq=self.freq)
+        future_ts = TSDataset(df=future_dataset, freq=self.freq)
+        future_ts.known_future = self.regressors
```
Comment (Mr-Geekman) [inline etna/datasets/tsdataset.py:282]: It seems like we can pass known_future directly in constructor of `TSDataset`.

File: etna/datasets/tsdataset.py [hunk 2]
```diff
@@ -278,7 +279,9 @@ def make_future(self, future_steps: int) -> "TSDataset":
 
         future_dataset = df.tail(future_steps).copy(deep=True)
         future_dataset = future_dataset.sort_index(axis=1, level=(0, 1))
-        future_ts = TSDataset(future_dataset, freq=self.freq)
+        future_ts = TSDataset(df=future_dataset, freq=self.freq)
+        future_ts.known_future = self.known_future
```
Comment (Mr-Geekman) [inline etna/datasets/tsdataset.py:283]: Can't we put this known future into the constructor of TSDataset on the previous line?

Comment (alex-hse-repository) [reply etna/datasets/tsdataset.py:283]: We can't, because the `_check_known_future` fails with `df_exog=Null`

Comment (Mr-Geekman) [reply etna/datasets/tsdataset.py:283]: May be a comment should be added about this

**Review by Mr-Geekman — APPROVED** [review]

**Pull request merged**